lib/internal/repl.js

```diff
@@ -93,7 +93,7 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
   var writing = false;
   var pending = false;
   repl.pause();
-  fs.open(historyPath, 'a+', oninit);
+  fs.open(historyPath, 'a+', 0o0600, oninit);
```

Comment:
Nit: Would you find adding a quick code comment with a quick explanation here. Nothing complicated, just enough for someone scanning through to know why the flag was set to this.